### PR TITLE
[5.1] Updating php-cs-fixer and php_codesniffer to latest versions

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
This updates php-cs-fixer and php_codesniffer to the latest versions respectively. I'd like to do this now since that removes a deprecated package.

### Testing Instructions
Everything should behave like before.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by richard67** [comment]
@Hackwar The composer.lock shows much more being updated than only what the description of this PR says. Is that the result of a plain `composer update` statement? This means you update all dependencies. Or have you explicitly only made a  `composer update friendsofphp/php-cs-fixer`?

**Comment by Hackwar** [comment]
I did a `composer update friendsofphp/php-cs-fixer --with-all-dependencies`, since otherwise it doesn't update. That is also the reason why this is aimed at 5.1-dev and not 5.0-dev.

**Comment by Hackwar** [comment]
I merged the necessary codestyle changes into this PR as well. No need to keep 2 PRs open for this...

**Comment by richard67** [comment]
@Hackwar @brianteeman 's comment in the other, closed PR is still valid for this PR here, and I agree with him: https://github.com/joomla/joomla-cms/pull/42604#issuecomment-1876825931 .

For "elseif" or "case" statements, the comments are before the corresponding "elseif" or case". The CS fixer now intends them by 4 more spaces because it assumes that these comments are referring to the actual "if" or "elseif" or "case" and not the one starting in the line below.

E.g. for "elseif" in file administrator/components/com_categories/src/Model/CategoriesModel.php in line 267 and 268:
```
            // Case: Using only the by level filter
        } elseif ($level) {
```

Or for the "case" in file administrator/components/com_finder/src/Indexer/Adapter.php in lines 929 and 930:

```
                // Archived items should only show up when option is enabled
            case 2:
                if ($this->params->get('search_archived', 1) == 0) {
```

It would be better readable if we would move these comments to the line below where they belong to, e.g. for "elseif" in file administrator/components/com_categories/src/Model/CategoriesModel.php:
```
        } elseif ($level) {
            // Case: Using only the by level filter
```

Or for the "case" in file administrator/components/com_finder/src/Indexer/Adapter.php:

```
            case 2:
                // Archived items should only show up when option is enabled
                if ($this->params->get('search_archived', 1) == 0) {
```

That's just a suggestion, I would not insist on it, and I know it requires some manual work.

But I think the better readability would make it be worth the work.

File: administrator/components/com_installer/src/Model/UpdatesitesModel.php [hunk 1]
```diff
@@ -631,12 +631,12 @@ protected function getListQuery()
                     $supportedIDs = InstallerHelper::getDownloadKeySupportedSites($enabled);
                     break;
 
-                // Show Update Sites which are missing Download Keys
+                    // Show Update Sites which are missing Download Keys
```
Comment (Quy) [inline administrator/components/com_installer/src/Model/UpdatesitesModel.php:634]: Move lines 629, 634, and 639.

File: plugins/content/pagenavigation/src/Extension/PageNavigation.php [hunk 2]
```diff
@@ -86,20 +86,21 @@ public function onContentBeforeDisplay($context, &$row, &$params, $page = 0)
...
                 switch ($orderDate) {
-                    // Use created if modified is not set
                     case 'modified':
+                        // Use created if modified is not set
                         $orderby = 'CASE WHEN ' . $db->quoteName('a.modified') . ' IS NULL THEN ' .
                             $db->quoteName('a.created') . ' ELSE ' . $db->quoteName('a.modified') . ' END';
                         break;
 
-                    // Use created if publish_up is not set
+
```
Comment (Quy) [inline plugins/content/pagenavigation/src/Extension/PageNavigation.php]:
```suggestion
```

**Comment by Quy** [comment]
Fix phpcs https://ci.joomla.org/joomla/joomla-cms/72832/1/7

**Comment by LadySolveig** [comment]
[PR phpcs fix](https://github.com/Hackwar/joomla-cms/pull/61)

[PR php-cs-fixer configuration update - deprecated rule](https://github.com/Hackwar/joomla-cms/pull/62)
I can also do a completely separate PR for this, but I actually think it's part of the update.

File: administrator/components/com_installer/src/Model/UpdatesitesModel.php [hunk 3]
```diff
@@ -631,12 +631,12 @@ protected function getListQuery()
...
                 case -1:
                     $supportedIDs = InstallerHelper::getDownloadKeyExistsSites(false, $enabled);
                     break;
 
-                // Show Update Sites which have valid Download Keys
+                    // Show Update Sites which have valid Download Keys
                 case 2:
                     $supportedIDs = InstallerHelper::getDownloadKeyExistsSites(true, $enabled);
```
Comment (Quy) [inline administrator/components/com_installer/src/Model/UpdatesitesModel.php]:
```suggestion
                case 2:
                    // Show Update Sites which have valid Download Keys
                    $supportedIDs = InstallerHelper::getDownloadKeyExistsSites(true, $enabled);
```

File: administrator/components/com_installer/src/Model/UpdatesitesModel.php [hunk 4]
```diff
@@ -631,12 +631,12 @@ protected function getListQuery()
                     $supportedIDs = InstallerHelper::getDownloadKeySupportedSites($enabled);
                     break;
 
-                // Show Update Sites which are missing Download Keys
+                    // Show Update Sites which are missing Download Keys
                 case -1:
                     $supportedIDs = InstallerHelper::getDownloadKeyExistsSites(false, $enabled);
```
Comment (Quy) [inline administrator/components/com_installer/src/Model/UpdatesitesModel.php]:
```suggestion
                case -1:
                    // Show Update Sites which are missing Download Keys
                    $supportedIDs = InstallerHelper::getDownloadKeyExistsSites(false, $enabled);
```

File: administrator/components/com_installer/src/Model/UpdatesitesModel.php [hunk 5]
```diff
@@ -631,12 +631,12 @@ protected function getListQuery()
                     $supportedIDs = InstallerHelper::getDownloadKeySupportedSites($enabled);
```
Comment (Quy) [inline administrator/components/com_installer/src/Model/UpdatesitesModel.php:631]: Move line 629 to here,

Comment (LadySolveig) [reply administrator/components/com_installer/src/Model/UpdatesitesModel.php:631]: Thank you!

File: administrator/components/com_installer/src/Model/UpdatesitesModel.php (hunk 3 above)
Comment (LadySolveig) [reply administrator/components/com_installer/src/Model/UpdatesitesModel.php]: Thanks

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 976031207372f83a0c07260a4dd152249d4330cf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42603">issues.joomla.org/tracker/joomla-cms/42603</a>.</sub>

**Review by wilsonge — APPROVED** [review]
Code review looks fine. I think if either of the symfony upgrades did anything we'd have bigger problems plus system tests would pick it up as they're much deeper libraries

**Comment by laoneo** [comment]
Just a not here, I did always do the `composer update` and then `composer bump`. Like that are the versions in the composer.json also updated.
As CS Fixer released 3.47.1 yesterday, you can update it already or we can do it then in beta.

**Comment by brianteeman** [comment]
Not addressing https://github.com/joomla/joomla-cms/pull/42603#issuecomment-1878492404 ??

**Comment by richard67** [comment]
> Not addressing [#42603 (comment)](https://github.com/joomla/joomla-cms/pull/42603#issuecomment-1878492404) ??

It was partly addressed. Some "case" statements have been fixed, but none of the "elseifs" or "else".

**Comment by richard67** [comment]
@LadySolveig Since your last commit, this PR has conflicts. And the issue which I had mentioned with my big comment has been addressed only partly, see my previous comment.

**Comment by LadySolveig** [comment]
@Richard it is not very clear to me why you did not address this when @Razzo1987 contacted you directly when he has done the manually clean up here 620d3e078aba40899614cdb834b7ab0d02c17371

could you please clean up the the remaining comments for the `elseif` and `else` yourself @Hackwar

**Comment by richard67** [comment]
> @richard it is not very clear to me why you did not address this when @Razzo1987 contacted you directly when he has done the manually clean up here [620d3e0](https://github.com/joomla/joomla-cms/commit/620d3e078aba40899614cdb834b7ab0d02c17371)

@LadySolveig Well I saw it was dealing only with the “case” statements so I assumed the other things will be done later. My first comment was very clear, I think.

**Comment by brianteeman** [comment]
Unfortunately doing it directly without a pr (not good practice) has already been partially undone by subsequent pull requests eg #42667

**Comment by LadySolveig** [comment]
I am not sure if I have understood you correctly, but I resolved the conflicts yesterday and made sure that the fixes for the comments remain as @Razzo1987  has already fixed them for us. @brianteeman have I overlooked something here somewhere?

@richard67 I had the impression that we agreed here in the discussion with the maintainers that it would be easier to fix this directly here in the PR. But if the opinion has changed here I will merge and we will fix the rest later as you suggested in your comment.

**Comment by richard67** [comment]
@LadySolveig Sorry in case if I’ve caused some confusion. Yes, it would be easier to fix it now, but we can do that also later. I don’t insist in my suggestions and am ok with merging the PR.

File: administrator/components/com_admin/src/Model/SysinfoModel.php [hunk 6]
```diff
@@ -355,7 +355,7 @@ private function getCompatPluginParameters()
     public function phpinfoEnabled(): bool
     {
         // remove any spaces from the ini value before exploding it
-        $disabledFunctions = str_replace(' ', '', ini_get('disable_functions'));
+        $disabledFunctions = str_replace(' ', '', \ini_get('disable_functions'));
```
Comment (HLeithner) [inline administrator/components/com_admin/src/Model/SysinfoModel.php:358]: `str_replace` will not be escaped?

File: administrator/components/com_finder/src/Indexer/Parser/Html.php [hunk 7]
```diff
@@ -38,7 +38,7 @@ class Html extends Parser
     public function parse($input)
     {
         // Strip invalid UTF-8 characters.
-        $oldSetting = ini_get('mbstring.substitute_character');
+        $oldSetting = \ini_get('mbstring.substitute_character');
         ini_set('mbstring.substitute_character', 'none');
```
Comment (HLeithner) [inline administrator/components/com_finder/src/Indexer/Parser/Html.php:42]: not so clear to me why `ini_get` get a `\` and `ini_set` not?

File: composer.json [hunk 8]
```diff
@@ -75,10 +75,10 @@
     "paragonie/sodium_compat": "^1.20",
     "phpmailer/phpmailer": "^6.8.1",
     "psr/link": "~1.1.1",
-    "symfony/console": "^6.3.4",
+    "symfony/console": "^6.4.2",
```
Comment (HLeithner) [inline composer.json:78]: why is a symfony/console and options-resolver update done in this pr? seems unrelated.

**Review by HLeithner — COMMENTED** [review]
rest looks ok by code review

File: administrator/components/com_admin/src/Model/SysinfoModel.php (hunk 6 above)
Comment (Hackwar) [reply administrator/components/com_admin/src/Model/SysinfoModel.php:358]: This is the result of php-cs-fixer in our current configuration. I only applied the necessary changes for our current configuration to pass again.

File: administrator/components/com_finder/src/Indexer/Parser/Html.php (hunk 7 above)
Comment (Hackwar) [reply administrator/components/com_finder/src/Indexer/Parser/Html.php:42]: It is the changed configuration of php-cs-fixer.

File: composer.json (hunk 8 above)
Comment (Hackwar) [reply composer.json:78]: I ran composer update with all necessary dependency updates. Before these changes, the dependencies did not resolve.

**Comment by Hackwar** [comment]
In the meantime, both codestyle packages have released new versions...

**Comment by LadySolveig** [comment]
Thank you !
